core/model/modx/moddashboardwidget.class.php

```diff
@@ -311,26 +313,14 @@ public function getFileChunk($tpl,array $placeholders = array()) {
      * Render the widget content as if it were a Snippet
      *
      * @param string $content
-     *
      * @return string
      */
-    public function renderAsSnippet($content = '')
-    {
-        if (empty($content)) {
-            $content = $this->widget->get('content');
-        }
-        $content = str_replace(['<?php', '?>'], '', $content);
-        $closure = function ($scriptProperties) use ($content) {
-            global $modx;
-            if (is_array($scriptProperties)) {
-                extract($scriptProperties, EXTR_SKIP);
-            }
-
-            return eval($content);
-        };
-
-        return $closure([
+    public function renderAsSnippet($content = '') {
+        if (empty($content)) $content = $this->widget->get('content');
+        $content = str_replace(array('<?php','?>'),'',$content);
+        $closure = create_function('$scriptProperties','global $modx;if (is_array($scriptProperties)) {extract($scriptProperties, EXTR_SKIP);}'.$content);
```

Comment:
Why here `$scriptProperties` variable in quotes? This code will fail. Variable dereferencing is possible only in double-quoted strings (ex. "$var").
Second issue - `create_function` is deprecated since 7.2 (http://php.net/manual/en/function.create-function.php).